src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/DefaultHtmlFactsService.cs

```diff
@@ -46,56 +46,75 @@ public override bool TryGetElementInfo(SyntaxNode element, out SyntaxToken conta
+            switch (attribute)
             {
-                prefixLocation = minimizedAttributeBlock.NamePrefix.Span;
-                selectedAttributeName = minimizedAttributeBlock.Name.GetContent();
-                selectedAttributeNameLocation = minimizedAttributeBlock.Name.Span;
-                return true;
-            }
-            else if (attribute is MarkupAttributeBlockSyntax attributeBlock)
-            {
-                prefixLocation = attributeBlock.NamePrefix.Span;
-                selectedAttributeName = attributeBlock.Name.GetContent();
-                selectedAttributeNameLocation = attributeBlock.Name.Span;
-                return true;
-            }
-            else if (attribute is MarkupTagHelperAttributeSyntax tagHelperAttribute)
-            {
-                prefixLocation = tagHelperAttribute.NamePrefix.Span;
-                selectedAttributeName = tagHelperAttribute.Name.GetContent();
-                selectedAttributeNameLocation = tagHelperAttribute.Name.Span;
-                return true;
-            }
-            else if (attribute is MarkupMinimizedTagHelperAttributeSyntax minimizedAttribute)
-            {
-                prefixLocation = minimizedAttribute.NamePrefix.Span;
-                selectedAttributeName = minimizedAttribute.Name.GetContent();
-                selectedAttributeNameLocation = minimizedAttribute.Name.Span;
-                return true;
-            }
-            else if (attribute is MarkupTagHelperDirectiveAttributeSyntax tagHelperDirectiveAttribute)
-            {
-                prefixLocation = tagHelperDirectiveAttribute.NamePrefix.Span;
-                selectedAttributeName = tagHelperDirectiveAttribute.FullName;
-                var fullNameSpan = TextSpan.FromBounds(tagHelperDirectiveAttribute.Transition.Span.Start, tagHelperDirectiveAttribute.Name.Span.End);
-                selectedAttributeNameLocation = fullNameSpan;
-                return true;
-            }
-            else if (attribute is MarkupMinimizedTagHelperDirectiveAttributeSyntax minimizedTagHelperDirectiveAttribute)
-            {
-                prefixLocation = minimizedTagHelperDirectiveAttribute.NamePrefix.Span;
-                selectedAttributeName = minimizedTagHelperDirectiveAttribute.FullName;
-                var fullNameSpan = TextSpan.FromBounds(minimizedTagHelperDirectiveAttribute.Transition.Span.Start, minimizedTagHelperDirectiveAttribute.Name.Span.End);
-                selectedAttributeNameLocation = fullNameSpan;
-                return true;
-            }
-            else if (attribute is MarkupMiscAttributeContentSyntax)
-            {
-                prefixLocation = null;
-                selectedAttributeName = null;
-                selectedAttributeNameLocation = null;
-                return true;
+                case MarkupMinimizedAttributeBlockSyntax minimizedAttributeBlock:
+                    if (minimizedAttributeBlock.NamePrefix == null)
```

Comment:
Why do we need to add these checks here in TryGetAttributeInfo? I'd imagine the caller would inspect the `prefixLocation` to see if it had any length to accomplish the same feat.